app/helpers/application_helper.rb

```diff
@@ -20,12 +20,23 @@ def icon(name, opts = {})
 
     if title
       direction = opts.delete(:tooltip) || 'bottom'
+      data = opts.delete(:data) || {}
 
-      content_tag(:div, class: classes + " tooltip tooltip-#{direction} hover:tooltip-open", style: "width: #{size}px; height: #{size}px;", data: { tip: title.to_s }, **opts.except(:class, :size, :variant)) do
+      content_tag(:div,
+        class: classes + " tooltip tooltip-#{direction} hover:tooltip-open",
+        style: "width: #{size}px; height: #{size}px;",
+        data: { tip: title.to_s }.merge(data),
+        **opts.except(:class, :size, :variant)
+      ) do
```

Comment:
I discovered a bug with this helper method: if you pass `title: ".."` and also pass `data-*` then the attributes were not getting properly constructed